lib/storage/bucket.js

```diff
@@ -349,6 +351,142 @@ Bucket.prototype.setMetadata = function(metadata, callback) {
+ *     to every file in the bucket as well. Default is false.
+ * @param {boolean}  options.force - By default, makePublic will return at the
+ *     first error found, but set this to true to continue processing if this
+ *     occurs.
+ * @param {Function} callback - The callback function.
+ *
+ * @example
+ * bucket.makePublic({ recursive: true }, function(err, files) {
+ *   // err is the first error to occur, otherwise null.
+ *   // files is an array of files successfully made public in the bucket.
+ * });
+ */
+Bucket.prototype.makePublic = function(options, callback) {
+  if (util.is(options, 'function')) {
+    callback = options;
+    options = {};
+  }
+
+  options.public = true;
+  this._makePublicPrivate(options, callback);
+};
+
+Bucket.prototype.makePrivate = function(options, callback) {
+  if (util.is(options, 'function')) {
+    callback = options;
+    options = {};
+  }
+
+  options.private = true;
+  this._makePublicPrivate(options, callback);
+};
+
+Bucket.prototype._makePublicPrivate = function(options, callback) {
+  var self = this;
+  var errors = [];
+  var filesProcessed = [];
+  var nextQuery = {};
+
+  // Allow public to read bucket contents
+  // while preserving original permissions
+  if (options.public) {
+    this.acl.add({
+      entity: 'allUsers',
+      role: 'READER'
+    }, processAllFiles);
+  } else if (options.private) {
+    var query = {
+      predefinedAcl: options.strict ? 'private' : 'projectPrivate'
+    };
+
+    // You aren't allowed to set both predefinedAcl & acl properties on a bucket
+    // so acl must explicitly be nullified.
+    var metadata = { acl: null };
+
+    this.makeReq_('PATCH', '', query, metadata, function(err, resp) {
+      if (err) {
+        processAllFiles(err);
+        return;
+      }
+      self.metadata = resp;
+      processAllFiles(null);
+    });
+  }
+
+  function processAllFiles(err) {
+    if (err) {
+      callback(err);
+      return;
+    }
+
+    if (options.recursive) {
+      async.whilst(function() {
+        // while there are still more files to make public
+        return !!nextQuery;
+      }, function(cb) {
+        processMoreFiles(nextQuery, cb);
+      }, function(err) {
+        if (err || errors.length) {
+          callback(err || errors, filesProcessed);
+        } else {
+          callback(null, filesProcessed);
+        }
+      });
+    } else {
+      callback(null, filesProcessed);
+    }
+  }
+
+  function processMoreFiles(query, callback) {
+    self.getFiles(query, function(err, files, nextPageQuery) {
+      if (err) {
+        callback(err);
+        return;
+      }
+
+      // Iterate through each file and make it public.
+      async.eachLimit(files, 10, function(file, cb) {
+        if (options.public) {
+          file.makePublic(processedCallback);
+        } else if (options.private) {
+          file.makePrivate({ strict: options.strict }, processedCallback);
+        }
+
+        function processedCallback(err) {
```

Comment:
Just leave it. Micro optimizations are generally worthless (unless the DOM is concerned) and just lead to weird code in JS.